File: app/src/main/java/com/oom/masterzuo/utils/LocalDisplay.java
package com.oom.masterzuo.utils;

import android.util.DisplayMetrics;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

public class LocalDisplay {

    public static int SCREEN_WIDTH_PIXELS;
    public static int SCREEN_HEIGHT_PIXELS;
    public static float SCREEN_DENSITY;
    public static int SCREEN_WIDTH_DP;
    public static int SCREEN_HEIGHT_DP;

    public static void init(DisplayMetrics dm) {

        SCREEN_WIDTH_PIXELS = dm.widthPixels;
        SCREEN_HEIGHT_PIXELS = dm.heightPixels;
        SCREEN_DENSITY = dm.density;
        SCREEN_WIDTH_DP = (int) (SCREEN_WIDTH_PIXELS / dm.density);
        SCREEN_HEIGHT_DP = (int) (SCREEN_HEIGHT_PIXELS / dm.density);
        Log.d("LOCAL_DISPLAY", "INIT APP " + " width: " + LocalDisplay.SCREEN_WIDTH_PIXELS +
                " height: " + LocalDisplay.SCREEN_HEIGHT_PIXELS);
    }

    public static int dp2px(float dp) {
        final float scale = SCREEN_DENSITY;
        return (int) (dp * scale + 0.5f);
    }

    public static int designedDP2px(float desingDP) {
        if (SCREEN_WIDTH_DP != 320) {
            desingDP = desingDP * SCREEN_WIDTH_DP / 320f;
        }
        return dp2px(desingDP);
    }

    public static void setPadding(final View view, float left, float top, float right, float bottom) {
        view.setPadding(designedDP2px(left), dp2px(top), designedDP2px(right), dp2px(bottom));
    }

    public static void setMargins(final View view, float left, float top, float right, float bottom) {
        LinearLayout.LayoutParams lllp = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
        lllp.setMargins(designedDP2px(left), (int)(top), designedDP2px(right), dp2px(bottom));
        view.setLayoutParams(lllp);
    }
}

File: app/src/main/java/com/masterzuo/masterzuo/utils/LocalDisplay.kt
package com.masterzuo.masterzuo.utils

import android.util.DisplayMetrics
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout

/**
 * Created by YoungDroid on 17/3/20.
 * Email: YoungDroid@163.com
 */
object LocalDisplay {

    var SCREEN_WIDTH_PIXELS: Int = 0
    var SCREEN_HEIGHT_PIXELS: Int = 0
    var SCREEN_DENSITY: Float = 0.toFloat()
    var SCREEN_WIDTH_DP: Int = 0
    var SCREEN_HEIGHT_DP: Int = 0

    fun init(dm: DisplayMetrics) {

        SCREEN_WIDTH_PIXELS = dm.widthPixels
        SCREEN_HEIGHT_PIXELS = dm.heightPixels
        SCREEN_DENSITY = dm.density
        SCREEN_WIDTH_DP = (SCREEN_WIDTH_PIXELS / dm.density).toInt()
        SCREEN_HEIGHT_DP = (SCREEN_HEIGHT_PIXELS / dm.density).toInt()
        debug("INIT APP " + " width: " + SCREEN_WIDTH_PIXELS + " height: " + SCREEN_HEIGHT_PIXELS)
    }

    fun dp2px(dp: Float): Int {
        val scale = SCREEN_DENSITY
        return (dp * scale + 0.5f).toInt()
    }

    fun designedDP2px(designDP: Float): Int {
        if (SCREEN_WIDTH_DP != 320) {
            dp2px(designDP * SCREEN_WIDTH_DP / 320f)
        }
        return dp2px(designDP)
    }

    fun setPadding(view: View, left: Float, top: Float, right: Float, bottom: Float) {
        view.setPadding(designedDP2px(left), dp2px(top), designedDP2px(right), dp2px(bottom))
    }

    fun setMargins(view: View, left: Float, top: Float, right: Float, bottom: Float) {
        val lllp = LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
        lllp.setMargins(designedDP2px(left), top.toInt(), designedDP2px(right), dp2px(bottom))
        view.layoutParams = lllp
    }
}
